src/schedule.cpp

```diff
@@ -239,6 +239,18 @@ struct stream_info
                 }
             }
         }
+
+        // move dangling parameter to the front so as not be removed
+        auto ins = std::next(last);
+        while(ins != p.end())
+        {
+            auto next = std::next(ins);
+            if(ins->name() == "@param")
+            {
+                p.move_instruction(ins, p.begin());
+            }
+            ins = next;
```

Comment:
why can't we remove `next` and just do `inst = std::next(ins)`?

Reply:
After we do the move_instruction, ins is the first element in the list, which is invalid for the while loop.